### PR TITLE
Rename TPR to CRD

### PR DESCRIPTION
**Pull request by nikhita**
TPR has been completely removed in 1.8. However, there are still some places which use the word "TPR".

We should rename TPR to CRD in such places to be consistent with the nomenclature.

**Release note**:

```release-note
NONE
```

/cc @sttts

### CODE REVIEW

File: hack/make-rules/test-cmd-util.sh [hunk 1]
```diff
@@ -1542,16 +1541,16 @@ run_non_native_resource_tests() {
   kube::log::status "Testing kubectl non-native resources"
   kube::util::non_native_resources
 
-  # Test that we can list this new third party resource (foos)
+  # Test that we can list this new CustomResource (foos)
```
Comment (resouer) [inline hack/make-rules/test-cmd-util.sh]: Why not custom resource definition? (not compulsory, just curious what's the formal name we are using)

Comment (idvoretskyi) [reply hack/make-rules/test-cmd-util.sh]: +1. It's better to use the full name.

Comment (nikhita) [reply hack/make-rules/test-cmd-util.sh]: This does a `kubectl get foos` which lists the _instances_ of our custom type i.e. it lists the "custom resources". [This](https://github.com/nikhita/kubernetes/blob/694630724a1e5036d35ed9c12dc97dd2989d15d6/hack/testdata/CRD/foo.yaml) is an instance of type `Foo` .

For the `CustomResourceDefinition` (where we register the type `Foo`), we need to do `kubectl get crd`, which is done [here](https://github.com/nikhita/kubernetes/blob/694630724a1e5036d35ed9c12dc97dd2989d15d6/hack/make-rules/test-cmd-util.sh#L1480).

Comment (resouer) [reply hack/make-rules/test-cmd-util.sh]: So it seems to be mixed usage or inconsistency?

Comment (sttts) [reply hack/make-rules/test-cmd-util.sh]: CRDs and CRs are two different things.

**Pull request merged**